x-pack/plugins/security/server/authentication/authentication_service.test.ts

```diff
@@ -329,6 +356,221 @@ describe('AuthenticationService', () => {
       });
     });
 
+    describe('unauthorized error handler', () => {
+      let unauthorizedErrorHandler: UnauthorizedErrorHandler;
+      let reauthenticate: jest.SpyInstance<Promise<AuthenticationResult>, [KibanaRequest]>;
+      let mockUnauthorizedErrorToolkit: jest.Mocked<UnauthorizedErrorHandlerToolkit>;
+      beforeEach(() => {
+        mockUnauthorizedErrorToolkit = { notHandled: jest.fn(), retry: jest.fn() };
+
+        service.start(mockStartAuthenticationParams);
+
+        unauthorizedErrorHandler =
+          mockSetupAuthenticationParams.elasticsearch.setUnauthorizedErrorHandler.mock.calls[0][0];
+        reauthenticate =
+          jest.requireMock('./authenticator').Authenticator.mock.instances[0].reauthenticate;
+      });
+
+      it('does not handle error if license is not available.', async () => {
+        mockSetupAuthenticationParams.license.isLicenseAvailable.mockReturnValue(false);
+
+        const failureReason = new errors.ResponseError(
+          securityMock.createApiResponse({ statusCode: 401, body: {} })
+        ) as UnauthorizedError;
+
+        await unauthorizedErrorHandler(
+          { error: failureReason, request: httpServerMock.createKibanaRequest() },
+          mockUnauthorizedErrorToolkit
+        );
+
+        expect(mockUnauthorizedErrorToolkit.notHandled).toHaveBeenCalledTimes(1);
+        expect(mockUnauthorizedErrorToolkit.retry).not.toHaveBeenCalled();
+        expect(reauthenticate).not.toHaveBeenCalled();
+      });
+
+      it('does not handle error when security is disabled in elasticsearch.', async () => {
```

Comment:
Question: this is just defensive coding, we don't really ever expect a 401 error if security is disabled in ES, right?

Reply:
Correct, just covering `if` code branches in the hook, but we shouldn't get 401 here. Unless security was disabled when request reached Kibana, then application handler was busy with something, in the meantime ES enabled security, and only then application handler decided to query Elasticsearch and obviously got `401` - 0.00001% chance :see_no_evil: